lightning/src/routing/gossip.rs

```diff
@@ -1433,6 +1433,22 @@ impl<L: Deref> NetworkGraph<L> where L::Target: Logger {
 			return Err(LightningError{err: "Channel announcement node had a channel with itself".to_owned(), action: ErrorAction::IgnoreError});
 		}
 
+		{
+			let channels = self.channels.read().unwrap();
+
+			if let Some(chan) = channels.get(&msg.short_channel_id) {
+				if chan.capacity_sats.is_some() {
```

Comment:
Is `capacity_sats` just one of the fields that would be present if we previously looked it up on-chain? Just wondering if there is something more explicit, but with the comment below I'd say it's pretty clear anyway.